example-gto/generate.sh

```diff
@@ -0,0 +1,186 @@
+#!/bin/bash
+
+# Setup script env:
+#   e   Exit immediately if a command exits with a non-zero exit status.
+#   u   Treat unset variables as an error when substituting.
+#   x   Print commands and their arguments as they are executed.
+set -eux
+
+PUSH=false
+echo $#
+if [ "$#" -eq 1 ] && [ "$0" != "--push" ]; then
+   PUSH=true;
+   echo "Will push things to GitHub :tada:"
+fi
+
+HERE="$( cd "$(dirname "$0")" ; pwd -P )"
+USER_NAME="iterative"
+REPO_NAME="example-gto"
+
+BUILD_PATH="$HERE/build"
+
+mkdir $BUILD_PATH
+pushd $BUILD_PATH
+if [ ! -d "$BUILD_PATH/.venv" ]; then
+  virtualenv -p python3 .venv
+  export VIRTUAL_ENV_DISABLE_PROMPT=true
+  source .venv/bin/activate
+  echo '.venv/' > .gitignore
+  pip install gitpython
+  pip install gto
+fi
+popd
+
+source $BUILD_PATH/.venv/bin/activate
+
+REPO_PATH="$HERE/build/$REPO_NAME"
+
+if [ -d "$REPO_PATH" ]; then
+  echo "Repo $REPO_PATH already exists, please remove it first."
+  exit 1
+fi
+
+TOTAL_TAGS=15
+STEP_TIME=100000
+BEGIN_TIME=$(( $(date +%s) - ( ${TOTAL_TAGS} * ${STEP_TIME}) ))
+export TAG_TIME=${BEGIN_TIME}
+export GIT_AUTHOR_DATE=${TAG_TIME}
+export GIT_COMMITTER_DATE=${TAG_TIME}
+tick(){
+  export TAG_TIME=$(( ${TAG_TIME} + ${STEP_TIME} ))
+  export GIT_AUTHOR_DATE=${TAG_TIME}
+  export GIT_COMMITTER_DATE=${TAG_TIME}
+}
+
+export GIT_AUTHOR_NAME="Alexander Guschin"
+export GIT_AUTHOR_EMAIL="1aguschin@gmail.com"
+export GIT_COMMITTER_NAME="$GIT_AUTHOR_NAME"
+export GIT_COMMITTER_EMAIL="$GIT_AUTHOR_EMAIL"
+
+
+
+mkdir -p $REPO_PATH
+pushd $REPO_PATH
+
+git init -b main
+cp -r $HERE/code/ .
+git add .
+tick
+git commit -m "Initialize Git repository with CI workflow"
+
+if $PUSH; then
+  # remove GH Actions workflows
+  gh api repos/$USER_NAME/$REPO_NAME/actions/runs \
+    --paginate -q '.workflow_runs[] | "\(.id)"' | \
+    xargs -n1 -I % gh api --silent repos/$USER_NAME/$REPO_NAME/actions/runs/% -X DELETE
+  # add remote
+  git remote add origin https://github.com/$USER_NAME/$REPO_NAME
+  # remove all tags from remote
+  git ls-remote --tags origin | awk '/^(.*)(\s+)(.*[a-zA-Z0-9])$/ {print ":" $2}' | xargs git push origin
+fi
+
+echo "Fix env"
+pip freeze > requirements.txt
```

Comment:
requirements file should be part of the code folder, `pip freeze` will produce a lot of things (probably all gto dependencies)

Reply:
not wrong, not a usual way (usually you expect just regular deps) and I would expect Pipfile.lock for this. I haven't seen this approach for the requirements for a while.

anyways, in this case it's also bad because you have to generate the new project anytime you update GTO, otherwise it's get stuck with the previous gto version

Reply:
Ok, fixing this.